lib/std/fs/path.zig

```diff
@@ -1319,3 +1319,40 @@ test "extension" {
     try testExtension("/foo/bar/bam/a.b.c", ".c");
     try testExtension("/foo/bar/bam/a.b.c/", ".c");
 }
+
+/// Returns the stem of the given file name (if any).
+/// Examples:
+/// - "hello/world/lib.tar.gz" ⇒ "lib.tar"
+/// - "hello/world/lib.tar"    ⇒ "lib"
+/// - "hello/world/lib"        ⇒ "lib"
```

Comment:
2 brief sentences what the stem is would be nice: The final path component, without its suffix. Shorthand for `self.basename(self.extension)`.

Reply:
> Shorthand for `self.basename(self.extension)`.

I don't know what you mean by that. Do you mean like `basename(extension("hello/world/lib.tar.gz"))`? That does not return "lib.tar", so it's not the same.

Reply:
Oh, my bad then. The docs of crystal are wrong then.

Reply:
They aren't wrong; notice how it works in Crystal: in `Path["/foo/bar/file.cr"].basename` the `Path["/foo/bar/file.cr"]` part is an additional `Path` struct they wrap the string in but in Zig we don't have any additional path at all so as the only argument to `basename` we take the actual path (simply a `[]const u8`) and then in Crystal they basically have a second argument which is a suffix (which I guess for us would do something like `std.mem.trimRight`) and it looks like `Path["foo/bar/baz.cr"].basename(".cr") == "baz"`, which we don't have. I don't think we should do what Crystal did. Our API is much simpler anyway.